optimizer/validator.go

```diff
@@ -78,6 +80,10 @@ func (v *validator) Leave(in ast.Node) (out ast.Node, ok bool) {
 		}
 	case *ast.PatternInExpr:
 		v.checkSameColumns(append(x.List, x.Expr)...)
+	case *ast.Limit:
+		if x.Count > math.MaxUint64-x.Offset {
```

Comment:
'`limit.Offset + limit.Count` is trimmed to MaxUint64 already. so there is no overflow.